### PR TITLE
Moddable city ranges

### PR DESCRIPTION
**Pull request by tuvus**
This PR addresses a popular request to modify the work ranges of cities #11422. Part of #3242.

These three things are made moddable:
- City bombard range, the default values is 2
- City work range, the default value is 3
- City expansion range, the default value is 5 (Is this right? Or should it be 6?)

I also want to support the ability for the work range to be higher than the expansion range.

I am not sure I found all of the various places in which it is used but I found the most obvious errors.
This feature will be experimental for a while and will surely have bugs. Luckily we're not forcing it on anyone.

<details><summary>Picture</summary>
City bombard range of 10

WorkRange3
WorkRange1
WorkRange2

</details>

### CODE REVIEW

**Comment by tuvus** [comment]
I added some pictures that I forgot.

File: core/src/com/unciv/ui/screens/cityscreen/CityScreen.kt [hunk 1]
```diff
@@ -351,8 +353,8 @@ class CityScreen(
         for (tileGroup in tileGroups) {
             val xDifference = city.getCenterTile().position.x - tileGroup.tile.position.x
             val yDifference = city.getCenterTile().position.y - tileGroup.tile.position.y
-            //if difference is bigger than 5 the tileGroup we are looking for is on the other side of the map
-            if (xDifference > 5 || xDifference < -5 || yDifference > 5 || yDifference < -5) {
+            //if difference is bigger than the expansion range the tileGroup we are looking for is on the other side of the map
+            if (xDifference > viewRange || xDifference < -viewRange || yDifference > viewRange || yDifference < -viewRange) {
```
[the remark this answers is not in the document]
Comment (tuvus) [reply core/src/com/unciv/ui/screens/cityscreen/CityScreen.kt:357]: Yeah, It seems to break world wrap maps.

[the remark this answers is not in the document]
Comment (tuvus) [reply core/src/com/unciv/ui/screens/cityscreen/CityScreen.kt:357]: Although, it still works with world wrap. Thats good.

File: core/src/com/unciv/logic/automation/civilization/NextTurnAutomation.kt [hunk 2]
```diff
@@ -490,10 +490,9 @@ object NextTurnAutomation {
             // Otherwise, AI tries to produce settlers when it can hardly sustain itself
             .filter { city ->
                 !workersBuildableForThisCiv
-                    || city.getCenterTile().getTilesInDistance(2).count { it.improvement!=null } > 1
-                    || city.getCenterTile().getTilesInDistance(3).any { it.civilianUnit?.hasUnique(UniqueType.BuildImprovements)==true }
-            }
-            .maxByOrNull { it.cityStats.currentCityStats.production }
+                    || city.getCenterTile().getTilesInDistance(civInfo.modConstants.cityWorkRange - 1 ).count { it.improvement!=null } > 1
+                    || city.getCenterTile().getTilesInDistance(civInfo.modConstants.cityWorkRange).any { it.civilianUnit?.hasUnique(UniqueType.BuildImprovements)==true }
```
[the remark this answers is not in the document]
Comment (tuvus) [reply core/src/com/unciv/logic/automation/civilization/NextTurnAutomation.kt:494]: I think I did as well, but didn't touch them.

File: core/src/com/unciv/ui/screens/cityscreen/CityScreen.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (tuvus) [reply core/src/com/unciv/ui/screens/cityscreen/CityScreen.kt:357]: > Depends on size. If we allow work radius = 42, the newgame code would need new checks to demand map radius > 42

From my testing, the work range can be higher than the map radius. With and without world wrap. Rectangle and hexegonal.

File: core/src/com/unciv/ui/screens/overviewscreen/WonderOverviewTab.kt [hunk 3]
```diff
@@ -244,12 +244,11 @@ class WonderInfo {
             }
             if (status == WonderStatus.NotFound && !knownFromQuest(viewingPlayer, name)) continue
             val city = if (status == WonderStatus.NotFound) null
-            else tile.getTilesInDistance(5)
-                .filter { it.isCityCenter() }
-                .filter { viewingPlayer.knows(it.getOwner()!!) }
-                .filter { viewingPlayer.hasExplored(it) }
-                .sortedBy { it.aerialDistanceTo(tile) }
-                .firstOrNull()?.getCity()
+            else gameInfo.getCities().filter { it.getCenterTile().aerialDistanceTo(tile) <= gameInfo.ruleset.modOptions.constants.cityExpandRange }
```
[the remark this answers is not in the document]
Comment (tuvus) [reply core/src/com/unciv/ui/screens/overviewscreen/WonderOverviewTab.kt]: That makes sense.

[the remark this answers is not in the document]
Comment (tuvus) [reply core/src/com/unciv/ui/screens/overviewscreen/WonderOverviewTab.kt]: > Nobody will notice anyway.

**Comment by yairm210** [comment]
This looks good to me, since it's fresh I won't add it in right now

**Review by yairm210 — APPROVED** [review]

**Pull request merged**